### PR TITLE
Mbed os cleanup

### PR DESCRIPTION
**Pull request by ATmobica**
#### Problem
Reference issue: #139 

 #### Summary of Changes
Remove install mbed-os to /opt directory
Cleanup chip-build-vscode docker image
Set MBED_OS_PATH variable in mbed_example.sh instead of override

I rebuilt mbed-os docker locally and checked that python requirements from mbed-os are not necessary to build our applications.

### CODE REVIEW

File: integrations/docker/images/chip-build-mbed-os/Dockerfile [hunk 1]
```diff
@@ -15,21 +15,10 @@ RUN set -x \
...
-    && wget --progress=dot:giga https://github.com/ARMmbed/mbed-os/archive/mbed-os-6.7.0.tar.gz \
-    && tar --strip-components=1 -xzf mbed-os-6.7.0.tar.gz \
-    && rm mbed-os-6.7.0.tar.gz) \
-    && : # last line
-
 # ------------------------------------------------------------------------------
 # Install Python modules
 RUN set -x \
     && pip3 install --no-cache-dir -U mbed-cli mbed-tools \
-    && pip3 install --no-cache-dir -r /opt/mbed-os/requirements.txt \
```
Comment (pan-) [inline integrations/docker/images/chip-build-mbed-os/Dockerfile:32]: How do we install mbed os requirements in the docker image ?

Comment (ATmobica) [reply integrations/docker/images/chip-build-mbed-os/Dockerfile:32]: Installing mbed-cli and mbed-tools should be enough for setup Python environment

Comment (pan-) [reply integrations/docker/images/chip-build-mbed-os/Dockerfile:32]: The system does not complain when we compile ? I was thinking mbed cli tool does not automaticaly install `mbed-os/requirements.txt`

Comment (ATmobica) [reply integrations/docker/images/chip-build-mbed-os/Dockerfile:32]: As far as I remember, there were no problems with that. I think the entire requirements for mbed-tools, which we directly use, are installed properly.

**Comment by lukgni** [comment]
Is this PR still valid or it is outdated?

**Comment by ATmobica** [comment]
It is rather outdated. But we probably still need some mbed-os cleanup after switching to third_party mbed-os and the latest version

**Comment by pan-** [comment]
@lukgni The change was OK for me and I would merge it if the removal of `pip3 install --no-cache-dir -r /opt/mbed-os/requirements.txt ` is double checked. Can you look at this ?

**Comment by lukgni** [comment]
@pan- Sure, I will take care of that.

**Comment by lukgni** [comment]
Ok, here are results (after applying this PR and rebuilding docker images):
- building/flashing [from tasks.json] and flashing/debugging [from launch.json] work properly inside chip-build-mbed-os container
- the same all workflows which based on our chip-build-mbed-os image

Unfortunately, I noticed that using chip-build-vscode doesn't work as we could expect. 
I guess that comes from fact that, for the long time we've used chip-build-mbed-os as a base inside .devcontainer/Dockerfile (instead of using chip-build-vscode image) so we didn't track all necessary requirements. 

Here are all issues:
- **'cortex-debug' extension expects to have 'arm-none-eabi-gdb' from mbed-os toolchain to be available within system PATH variable.** - necessary export has been done only in chip-build-mbed-os image definition: 
https://github.com/ARMmbed/connectedhomeip/blob/4ea61739bc382340bbfe34eb609f5858f8148eb1/integrations/docker/images/chip-build-mbed-os/Dockerfile#L56
In 'chip-build-vsode' case, the gdb is not available within PATH - I guess we will have to figure out how
 to deal with multiple toolchains in that image to solve that problem first. 
It looks like we can use `armToolchainPath` parameter inside 'cortex-debug' based tasks (launch.json) (https://marketplace.visualstudio.com/items?itemName=marus25.cortex-debug) 

- **openocd is not avaialble within system PATH variable** - well, this is the same case as above. To solve this we will need to modify all of our scripts where openocd is used to point to its direct location (`${OPENOCD_PATH}/bin/openocd`) like in mbed_example.sh:
https://github.com/ARMmbed/connectedhomeip/blob/4ea61739bc382340bbfe34eb609f5858f8148eb1/scripts/examples/mbed_example.sh#L137

**_All observed issues are not related to this PR and we may consider convert them to GitHub issues and address them later._**

**Review by lukgni — COMMENTED** [review]
@pan-  Ok, here are results (after applying this PR and rebuilding docker images):
- building/flashing [from tasks.json] and flashing/debugging [from launch.json] work properly inside chip-build-mbed-os container
- the same all workflows which based on our chip-build-mbed-os image

***

Unfortunately, I noticed that using `chip-build-vscode` doesn't work as we could expect. 
I guess that comes from fact that, for the long time we've used `chip-build-mbed-os` as a base inside .devcontainer/Dockerfile (instead of using chip-build-vscode image) so we didn't track all necessary requirements. 

Here are all issues:
- **'cortex-debug' extension expects to have 'arm-none-eabi-gdb' from mbed-os toolchain to be available within system PATH variable.** - necessary export has been done only in chip-build-mbed-os image definition: 
https://github.com/ARMmbed/connectedhomeip/blob/4ea61739bc382340bbfe34eb609f5858f8148eb1/integrations/docker/images/chip-build-mbed-os/Dockerfile#L56
In 'chip-build-vsode' case, the gdb is not available within PATH - I guess we will have to figure out how
 to deal with multiple toolchains in that image to solve that problem first. 
It looks like we can use `armToolchainPath` parameter inside 'cortex-debug' based tasks (launch.json) (https://marketplace.visualstudio.com/items?itemName=marus25.cortex-debug) 

- **openocd is not avaialble within system PATH variable** - well, this is the same case as above. To solve this we will need to modify all of our scripts where openocd is used to point to its direct location (`${OPENOCD_PATH}/bin/openocd`). Here is example where we are calling openocd from PATH variable:
https://github.com/ARMmbed/connectedhomeip/blob/4ea61739bc382340bbfe34eb609f5858f8148eb1/scripts/examples/mbed_example.sh#L137

***
**_All observed issues are not related to this PR and we may consider convert them to GitHub issues and address them later._**
 ***

**Review by lukgni — APPROVED** [review]

**Comment by ATmobica** [comment]
This PR is no longer valid. We will do the cleanup directly on the upstream docker image version. It contains part of the fixes from this PR. We just need to remove the Mbed OS sources and toolchain from it.

**Pull request closed without merging**